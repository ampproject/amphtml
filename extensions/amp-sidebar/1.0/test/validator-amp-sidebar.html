<!--
  Copyright 2017 The AMP HTML Authors. All Rights Reserved.

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS-IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the license.
-->
<!--
  Test Description:
  Tests support for the amp-sidebar tag.
-->
<!doctype html>
<html âš¡>
<head>
  <meta charset="utf-8">
  <link rel="canonical" href="./regular-html-version.html">
  <meta name="viewport" content="width=device-width,minimum-scale=1">
  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
  <script async src="https://cdn.ampproject.org/v0.js"></script>
  <script async custom-element="amp-accordion" src="https://cdn.ampproject.org/v0/amp-accordion-0.1.js"></script>
  <script async custom-element="amp-sidebar" src="https://cdn.ampproject.org/v0/amp-sidebar-0.1.js"></script>
</head>
<body>
  <!-- Valid: Example of amp-sidebar usage -->
  <amp-sidebar side="left" layout="nodisplay">
    <amp-accordion>
      <section>
        <h2>Section 1</h2>
        <p>Content</p>
      </section>
      <section>
        <h2>Section 2</h2>
        <p>Content</p>
      </section>
    </amp-accordion>
    <amp-img layout="fill" src="img"></amp-img>
  </amp-sidebar>
  <!-- Valid: Example of amp-sidebar with toolbar usage -->
  <amp-sidebar layout="nodisplay">
    <nav toolbar="(min-width: 0px) and (max-width: 768px)" toolbar-target="id">
      <ul>
        <li>Nav item 1</li>
        <li>Nav item 2</li>
        <li>Nav item 3</li>
      </ul>
    </nav>
    <amp-img layout="fill" src="img"></amp-img>
  </amp-sidebar>
  <!-- Invalid: more than one <ul> tag -->
  <amp-sidebar layout="nodisplay">
    <nav toolbar="(min-width: 0px) and (max-width: 768px)" toolbar-target="id">
      <ul>
        <li>Nav item 1</li>
      </ul>
      <ul>
        <li>Nav item 2</li>
        <li>Nav item 3</li>
      </ul>
    </nav>
    <amp-img layout="fill" src="img"></amp-img>
  </amp-sidebar>
  <!-- Invalid: disallowed children -->
  <amp-sidebar layout="nodisplay">
    <amp-ad layout="fill" src="https://ad" type="ad"></amp-ad>
  </amp-sidebar>
  <!-- Invalid: incorrect side value -->
  <amp-sidebar side="center" layout="nodisplay"></amp-sidebar>
</body>
</html>
