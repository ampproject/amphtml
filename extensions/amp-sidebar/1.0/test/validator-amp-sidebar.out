FAIL
amp-sidebar/1.0/test/validator-amp-sidebar.html:59:4 Tag 'amp-sidebar > nav' must have 1 child tags - saw 2 child tags. [AMP_TAG_PROBLEM]
amp-sidebar/1.0/test/validator-amp-sidebar.html:72:4 The tag 'amp-ad' may not appear as a descendant of tag 'amp-sidebar'. (see https://www.ampproject.org/docs/reference/components/amp-ad) [AMP_TAG_PROBLEM]
amp-sidebar/1.0/test/validator-amp-sidebar.html:75:2 The attribute 'side' in tag 'amp-sidebar' is set to the invalid value 'center'. (see https://www.ampproject.org/docs/reference/components/amp-sidebar) [AMP_TAG_PROBLEM]
